psi4/src/psi4/dfmp2/mp2.cc

```diff
@@ -2800,6 +2802,18 @@ void RDFMP2::form_gradient()
     }
     timer_off("Grad: V");
 
+    // If an external field exists, add it to the one-electron Hamiltonian
+    py::object pyExtern = dynamic_cast<PythonDataType*>(options_["EXTERN"].get())->to_python();
+    if (pyExtern) {
+        std::shared_ptr<ExternalPotential> external = pyExtern.cast<std::shared_ptr<ExternalPotential>>();
+        if (external) {
+            gradient_terms.push_back("External Potential");
+            timer_on("Grad: External");
+            gradients_["External Potential"] = external->computePotentialGradients(basisset_, PAO);
+            timer_off("Grad: External");
+        }  // end external
+    }
+
```

Comment:
Awesome, thanks Andy! As a note I guess this will effect all other gradients (and Hessians) within Psi4 as well? We might need to add an extra list to filter out gradients that have not considered external potentials.